install_cni_test.go

```diff
@@ -0,0 +1,188 @@
+package main_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+)
+
+var script = "k8s-install/scripts/install-cni.sh"
+
+var defaultEnvVars = map[string]string{
+	"KUBERNETES_SERVICE_HOST": "127.0.0.1",
+	"KUBERNETES_SERVICE_PORT": "8080",
+}
+
+// runCniScript will run install-cni.sh, and return the output of the command and an error
+func runCniScript() (string, error) {
+	result, err := exec.Command(script).CombinedOutput()
+	if err != nil {
+		return string(result), err
+	}
+	return string(result), nil
+}
+
+var _ = BeforeSuite(func() {
+	err := os.Setenv("SLEEP", "false")
+	if err != nil {
+		Fail("Failed to set env var 'SLEEP'")
+	}
+})
+
+var _ = Describe("install-cni.sh tests", func() {
+	BeforeEach(func() {
+		// Create the needed dirs.
+		err := os.MkdirAll("/host/opt/cni/bin", 755)
+		if err != nil {
+			Fail("Failed to create CNI bin dir")
+		}
+		err = os.MkdirAll("/host/etc/cni/net.d", 755)
+		if err != nil {
+			Fail("Failed to create CNI conf dir")
+		}
+
+		// Setup out env vars.
+		for k, v := range defaultEnvVars {
+			err := os.Setenv(k, v)
+			if err != nil {
+				Fail(fmt.Sprintf("Failed to set env var: %s", k))
+			}
+		}
+
+		// Move the temp config file into place.
+		err = exec.Command("cp", "k8s-install/scripts/calico.conf.default", "/calico.conf.tmp").Run()
+		if err != nil {
+			Fail("Failed to move default CNI conf file")
+		}
+	})
+
+	AfterEach(func() {
+		// Unset our env vars
+		for k, _ := range defaultEnvVars {
+			err := os.Unsetenv(k)
+			if err != nil {
+				Fail(fmt.Sprintf("Failed to unset env var %s", k))
+			}
+		}
+
+		// Remove the bins and conf file between each run.
+		err := os.RemoveAll("/host/opt/cni/bin/")
+		if err != nil {
+			Fail(fmt.Sprintf("%v", err))
+		}
+		err = os.RemoveAll("/host/opt/cni/bin/")
+		if err != nil {
+			Fail(fmt.Sprintf("%v", err))
+		}
+	})
+
+	Describe("Run install-cni", func() {
+		Context("With default values", func() {
+			It("Should install bins and config", func() {
+				_, err := runCniScript()
+				Expect(err).NotTo(HaveOccurred())
+
+				// Get a list of files in the defualt CNI bin location.
+				files, err := ioutil.ReadDir("/host/opt/cni/bin")
+				if err != nil {
+					Fail("Could not list the files in /host/opt/cni/bin")
+				}
+				names := []string{}
+				for _, file := range files {
+					names = append(names, file.Name())
+				}
+
+				// Get a list of files in the default location for CNI config.
+				files, err = ioutil.ReadDir("/host/etc/cni/net.d")
+				if err != nil {
+					Fail("Could not list the files in /host/etc/cni/net.d")
+				}
+				for _, file := range files {
+					names = append(names, file.Name())
+				}
+
+				Expect(names).To(ContainElement("calico"))
+				Expect(names).To(ContainElement("calico-ipam"))
+				Expect(names).To(ContainElement("10-calico.conf"))
+			})
+			It("Should parse and output a templated config", func() {
```

Comment:
I'd like to see a test for each supported variable and assert that:
- The right values get templated in
- The right default values (if any) are set.

Happy for that to be a follow-on PR once we get the basic structure right here though.